src/coreclr/tools/Common/TypeSystem/IL/Stubs/DelegateThunks.cs

```diff
@@ -496,7 +496,7 @@ public override MethodIL EmitIL()
             // We will generate the following code:
             //
             // object ret;
-            // object[] args = new object[parameterCount];
+            // object[] args = DelegateHelpers.GetObjectArray(parameterCount);
```

Comment:
Is it absolutely guaranteed that this array is not passed to user code that can keep it around? It is hard to review see all places where this array can flow in System.Linq.Expressions.

Reply:
This would be the input into the interpreter, which should not retain the array after the call. It would not be many places to review.  It is true though that allocating a temp buffer (vs. renting) is a lot simpler to reason about.

Reply:
If we cared about the interpreter performance, it would be best to change this to `Span<object>`. It would allow us to allocate the small sizes on stack and it would be correct by construction. (I understand that it would require matching changes in System.Linq.Expression as well.)